docs/architecture/cluster-registration-workflow.md

```diff
@@ -0,0 +1,40 @@
+# Cluster registration workflow
+
+Registering a cluster in Clusternet is similar to joining a new node to Kubernetes.
+
+The basic registration process is that the agent first sends a registration request to the hub cluster through the bootstrap token of the hub cluster. After the hub cluster detects the registration request, the hub cluster will start to create some necessary resources for registration. The registration request will be marked as approved after these resources to be created and ready.  When the cluster registration success, the agent will periodically send its heartbeat information to the hub cluster.
+
+## Architecture
+
+<img src="../images/cluster-registration-workflow.png" style="width:1000px;"/>
```

Comment:
Could you add the legend to the graph as well to explain the  type of the shape as well?